ACPs/224-dynamic-gas-limit-in-subnet-evm/README.md

```diff
@@ -0,0 +1,233 @@
+| K | initial gas price update factor | ~87*T | :white_check_mark: Through `TimeToDouble` (default 60s)
+
+#### Adjustment to ACP-176 calculations for price discovery
+
+// TODO: MinGasPrice requires a binary search to find the correct value (similar to how we do for DesiredTargetExcess in Coreth)
+
+### Genesis Configuration
+
+There will be a new genesis chain configuration to set the parameters for the chain without requiring the ACP176FeeManager precompile to be activated. This will be similar to the existing fee configuration parameters in chain configuration. If there is no genesis configuration for the new fee parameters the default values for C-Chain will be used. This will look like the following:
+
+```json
+{
+  ...
+  "acp224Timestamp": ...,
+  "acp224FeeConfig": {
+    "targetGas": ...,
+    "minGasPrice": ...,
+    "maxCapacityFactor": ...,
+    "timeToDouble": ...,
+    "targetBlockRate": ...,
+    "minBlockGasCost": ...,
+    "maxBlockGasCost": ...,
+    "blockGasCostStep": ...
+  }
+}
+
+```
+
+### New ACP176 Fee Manager Precompile
+
+A new fee manager precompile will be required to dynamically changing the parameters. The precompile will be similar to the existing fee manager precompile. The solidity interface will be as follows:
+
+```solidity
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+import "./IAllowList.sol";
+
+interface IACP176FeeManager is IAllowList {
+  struct ACP176FeeConfig {
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  }
+  event FeeConfigChanged(address indexed sender, ACP176FeeConfig oldFeeConfig, ACP176FeeConfig newFeeConfig);
+
+  // Set fee config fields to contract storage
+  function setFeeConfig(
+    uint256 targetGas;
+    uint256 minGasPrice;
+    uint256 maxCapacityFactor;
+    uint256 timeToDouble;
+    uint256 targetBlockRate;
+    uint256 minBlockGasCost;
+    uint256 maxBlockGasCost;
+    uint256 blockGasCostStep;
+  ) external;
+
+  // Get fee config from the contract storage
+  function getFeeConfig()
+    external
+    view
+    returns (
+      uint256 targetGas;
+      uint256 minGasPrice;
+      uint256 maxCapacityFactor;
+      uint256 timeToDouble;
+      uint256 targetBlockRate;
+      uint256 minBlockGasCost;
+      uint256 maxBlockGasCost;
+      uint256 blockGasCostStep;
+    );
+
+  // Get the last block number changed the fee config from the contract storage
+  function getFeeConfigLastChangedAt() external view returns (uint256 blockNumber);
+}
+```
+
+The precompile will have a `targetGas` parameter that is used to set the target gas consumed per second for ACP-176. This will override the `gas-target` in node config. If the new fee manager precompile is not activated, validators can keep using `gas-target` in node config to set the target gas consumed per second (as in C-Chain).
```

Comment:
Is following of the `targetGas` (or more technically `q` under the hood) set in the `ACP176FeeManager` precompile something that is enforced by block verification? 

For instance, primary network validators today can set their preferred target gas to any value they want, but the effect target gas value for the chain can only be altered by a fixed factor in each subsequent block, and the new effect target gas value being within the allowed bounds is verified in block verification.

For chains with `ACP176FeeManager` precompile activated, can we somehow just enforce that the target gas value is always equal to the latest accept value provided to the precompile? So if `setFeeConfig` is called in block N, block N+1 must have the effective `targetGas` specified in the final call in block N.